SI-6762 rename emptyValDef to emptySelfType.

paulp opened the pull request:
Looks like emptyValDef.isEmpty was already changed to return
false, so now all that's left is a name which means something.

paulp commented:
Review by @xeno-by

xeno-by commented:
How about `noSelfType`? That would avoid the potential confusion of predicting what `emptySelfType.isEmpty` should evaluate to.

paulp commented:
Another fine day at the productivity factory.

``` diff
 diff --git i/test/files/run/repl-power.check w/test/files/run/repl-power.check
 index 575d94211c..c1992e89aa 100644
 --- i/test/files/run/repl-power.check
 +++ w/test/files/run/repl-power.check
 @@ -31,4 +31,4 @@ m: $r.treedsl.global.Match =
  scala> typed(m).tpe                              // typed is in scope
  res2: $r.treedsl.global.Type = Boolean

 -scala>
 +scala> 
```

It doesn't display for me, but imagine that trailing space.

xeno-by commented on src/reflect/scala/reflect/api/Trees.scala:
```diff
@@ -72,12 +72,12 @@ trait Trees { self: Universe =>
      *
      *  Empty trees are: the `EmptyTree` null object, `TypeTree` instances that don't carry a type
-     *  and the special `emptyValDef` singleton.
+     *  and the special `noSelfType` singleton.
```
Looks like this comment can go.

xeno-by commented on src/reflect/scala/reflect/api/Trees.scala:
```diff
@@ -72,12 +72,12 @@ trait Trees { self: Universe =>
      *
-     *  Unfortunately `emptyValDef` is also considered to be `isEmpty`, but this is deemed to be
+     *  Unfortunately `noSelfType` is also considered to be `isEmpty`, but this is deemed to be
      *  a conceptual mistake pending a fix in https://issues.scala-lang.org/browse/SI-6762.
```
And this one too.

xeno-by commented on src/reflect/scala/reflect/api/Trees.scala:
```diff
@@ -2121,6 +2121,9 @@ trait Trees { self: Universe =>
    */
+  val noSelfType: ValDef
+
+  @deprecated("Use noSelfType", "2.11.0")
```
Looks like most of our deprecation messages say `Use ... instead`.

xeno-by commented on src/reflect/scala/reflect/internal/StdNames.scala:
```diff
@@ -661,7 +661,7 @@ trait StdNames {
     val drop: NameType                 = "drop"
     val elem: NameType                 = "elem"
-    val emptyValDef: NameType          = "emptyValDef"
+    val noSelfType: NameType        = "noSelfType"
```
Alignment got skewed.

paulp committed:
SI-6762 rename emptyValDef to noSelfType.

Looks like emptyValDef.isEmpty was already changed to return
false, so now all that's left is a name which means something.

paulp commented:
Since if you made me tweak this one more time I would be forced to fly to where you are and throttle you, I sense that this must LGTY.

xeno-by commented:
Yes, that was an lgtm.
